python312Packages.quackosm: init at 0.8.2; python312Packages.{pyarrow-ops,parametrization,geohash}, geoarrow-c: init

Sigmanificient opened the pull request:
## Description of changes

WIP fix for:
- #327117

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Sigmanificient committed:
python312Packages.geoarrow-rust-core: init at 0.2.0

style fix

Sigmanificient committed:
python312Packages.pyarrow-ops: init at 0.0.8

fix style

Sigmanificient committed:
geoarrow-c: init at 0.1.2

fix

Sigmanificient committed:
python312Packages.parametrization: init at 2021.4.1

style fix

Sigmanificient commented:
@ofborg build python312Packages.{geoparquet,python-parametrization,geoarrow-{pyarrow,c,pandas,rust-core},pyarrow-ops}

Sigmanificient commented:
@ofborg python312Packages.{geoparquet,python-parametrization,geoarrow-pyarrow,geoarrow-c,geoarrow-pandas,geoarrow-rust-core,pyarrow-ops}

Sigmanificient commented:
@ofborg build python312Packages.{geoparquet,python-parametrization,geoarrow-pyarrow,geoarrow-c,geoarrow-pandas,geoarrow-rust-core,pyarrow-ops}

Sigmanificient committed:
python312Packages.geohash: init at 1.0

top level geohash

Sigmanificient committed:
python312Packages.quackosm: init at 0.8.2

remove srai

quackosm

sikmir commented on pkgs/development/python-modules/geoarrow-c/default.nix:
```diff
@@ -0,0 +1,42 @@
+    hash = "sha256-uEB+D3HhrjnCgExhguZkmvYzULWo5gAWxXeIGQOssqo=";
+  };
+
+  sourceRoot = "${src.name}/python/${pname}";
```
```suggestion
  sourceRoot = "${src.name}/python/geoarrow-c";
```

sikmir commented on pkgs/development/python-modules/geoarrow-pandas/default.nix:
```diff
@@ -0,0 +1,47 @@
+    hash = "sha256-Ni+GKTRhRDRHip1us3OZPuUhHQCNU7Nap865T/+CU8Y=";
+  };
+
+  sourceRoot = "${src.name}/${pname}";
```
```suggestion
  sourceRoot = "${src.name}/geoarrow-pandas";
```

sikmir commented on pkgs/development/python-modules/h3ronpy/default.nix:
```diff
@@ -0,0 +1,40 @@
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  meta = {
+    description = "Data science toolkit for the H3 geospatial grid ";
```
```suggestion
    description = "Data science toolkit for the H3 geospatial grid";
```

sikmir commented:
What is wrong with geoarrow-rs?

Sigmanificient commented:
> What is wrong with geoarrow-rs?

Well currently it fails to build with `stdio.h` not found error

sikmir commented:
Please resolve folium conflict.

Sigmanificient commented:
This version of the derivation from [waffle8946](https://discourse.nixos.org/t/rustpackage-stuck-on-missing-header-stdio-h/49383/3) builds, but produces an empty output

Sigmanificient commented:
@sikmir I think 
```py
  cargoDeps = rustPlatform.importCargoLock {
    lockFile = "${src}/python/core/Cargo.lock";
  };
```
is causing an IFD? ofborg seems angry about it, any idea to solve this?

RaczeQ commented:
Hi, I'm the QuackOSM developer and I wanted to understand what sources did you use to get dependency tree for the library (pdm.lock?).

I see that you are struggling with srai / folium and other libraries that aren't necessary for the QuackOSM to function and they are here only for the development (tests, docs generation etc)

Here is the output of the `pipgrip` tool for real dependency tree:

Version without CLI
```
quackosm (0.8.2)
├── beautifulsoup4 (4.12.3)
│   └── soupsieve>1.2 (2.5)
├── duckdb>=0.10.2 (1.0.0)
├── geoarrow-pandas>=0.1.1 (0.1.1)
│   ├── geoarrow-pyarrow (0.1.2)
│   │   ├── geoarrow-c (0.1.2)
│   │   ├── pyarrow (17.0.0)
│   │   │   └── numpy>=1.16.6 (2.0.1)
│   │   └── pyarrow-hotfix (0.6)
│   ├── pandas (2.2.2)
│   │   ├── numpy>=1.22.4 (2.0.1)
│   │   ├── python-dateutil>=2.8.2 (2.9.0.post0)
│   │   │   └── six>=1.5 (1.16.0)
│   │   ├── pytz>=2020.1 (2024.1)
│   │   └── tzdata>=2022.7 (2024.1)
│   └── pyarrow (17.0.0)
│       └── numpy>=1.16.6 (2.0.1)
├── geoarrow-pyarrow>=0.1.2 (0.1.2)
│   ├── geoarrow-c (0.1.2)
│   ├── pyarrow (17.0.0)
│   │   └── numpy>=1.16.6 (2.0.1)
│   └── pyarrow-hotfix (0.6)
├── geoarrow-rust-core>=0.2.0 (0.2.0)
├── geopandas>=0.6 (1.0.1)
│   ├── numpy>=1.22 (2.0.1)
│   ├── packaging (24.1)
│   ├── pandas>=1.4.0 (2.2.2)
│   │   ├── numpy>=1.22.4 (2.0.1)
│   │   ├── python-dateutil>=2.8.2 (2.9.0.post0)
│   │   │   └── six>=1.5 (1.16.0)
│   │   ├── pytz>=2020.1 (2024.1)
│   │   └── tzdata>=2022.7 (2024.1)
│   ├── pyogrio>=0.7.2 (0.9.0)
│   │   ├── certifi (2024.7.4)
│   │   ├── numpy (2.0.1)
│   │   └── packaging (24.1)
│   ├── pyproj>=3.3.0 (3.6.1)
│   │   └── certifi (2024.7.4)
│   └── shapely>=2.0.0 (2.0.5)
│       └── numpy<3,>=1.14 (2.0.1)
├── polars>=0.19.4 (1.2.1)
├── pooch>=1.6.0 (1.8.2)
│   ├── packaging>=20.0 (24.1)
│   ├── platformdirs>=2.5.0 (4.2.2)
│   └── requests>=2.19.0 (2.32.3)
│       ├── certifi>=2017.4.17 (2024.7.4)
│       ├── charset-normalizer<4,>=2 (3.3.2)
│       ├── idna<4,>=2.5 (3.7)
│       └── urllib3<3,>=1.21.1 (2.2.2)
├── psutil>=5.6.2 (6.0.0)
├── pyarrow-ops (0.0.8)
│   ├── numpy>=1.19.2 (2.0.1)
│   └── pyarrow>=3.0 (17.0.0)
│       └── numpy>=1.16.6 (2.0.1)
├── pyarrow>=16.0.0 (17.0.0)
│   └── numpy>=1.16.6 (2.0.1)
├── requests (2.32.3)
│   ├── certifi>=2017.4.17 (2024.7.4)
│   ├── charset-normalizer<4,>=2 (3.3.2)
│   ├── idna<4,>=2.5 (3.7)
│   └── urllib3<3,>=1.21.1 (2.2.2)
├── rich>=10.11.0 (13.7.1)
│   ├── markdown-it-py>=2.2.0 (3.0.0)
│   │   └── mdurl~=0.1 (0.1.2)
│   └── pygments<3.0.0,>=2.13.0 (2.18.0)
├── shapely>=2 (2.0.5)
│   └── numpy<3,>=1.14 (2.0.1)
├── tqdm>=4.42.0 (4.66.4)
└── typeguard>=3.0.0 (4.3.0)
    └── typing-extensions>=4.10.0 (4.12.2)
```

Version with CLI
```
quackosm[cli] (0.8.2)
├── beautifulsoup4 (4.12.3)
│   └── soupsieve>1.2 (2.5)
├── duckdb>=0.10.2 (1.0.0)
├── geoarrow-pandas>=0.1.1 (0.1.1)
│   ├── geoarrow-pyarrow (0.1.2)
│   │   ├── geoarrow-c (0.1.2)
│   │   ├── pyarrow (17.0.0)
│   │   │   └── numpy>=1.16.6 (1.26.4)
│   │   └── pyarrow-hotfix (0.6)
│   ├── pandas (2.2.2)
│   │   ├── numpy>=1.22.4 (1.26.4)
│   │   ├── python-dateutil>=2.8.2 (2.9.0.post0)
│   │   │   └── six>=1.5 (1.16.0)
│   │   ├── pytz>=2020.1 (2024.1)
│   │   └── tzdata>=2022.7 (2024.1)
│   └── pyarrow (17.0.0)
│       └── numpy>=1.16.6 (1.26.4)
├── geoarrow-pyarrow>=0.1.2 (0.1.2)
│   ├── geoarrow-c (0.1.2)
│   ├── pyarrow (17.0.0)
│   │   └── numpy>=1.16.6 (1.26.4)
│   └── pyarrow-hotfix (0.6)
├── geoarrow-rust-core>=0.2.0 (0.2.0)
├── geopandas>=0.6 (0.14.4)
│   ├── fiona>=1.8.21 (1.9.6)
│   │   ├── attrs>=19.2.0 (23.2.0)
│   │   ├── certifi (2024.7.4)
│   │   ├── click-plugins>=1.0 (1.1.1)
│   │   │   └── click>=4.0 (8.1.7)
│   │   ├── click~=8.0 (8.1.7)
│   │   ├── cligj>=0.5 (0.7.2)
│   │   │   └── click>=4.0 (8.1.7)
│   │   └── six (1.16.0)
│   ├── numpy>=1.22 (1.26.4)
│   ├── packaging (24.1)
│   ├── pandas>=1.4.0 (2.2.2)
│   │   ├── numpy>=1.22.4 (1.26.4)
│   │   ├── python-dateutil>=2.8.2 (2.9.0.post0)
│   │   │   └── six>=1.5 (1.16.0)
│   │   ├── pytz>=2020.1 (2024.1)
│   │   └── tzdata>=2022.7 (2024.1)
│   ├── pyproj>=3.3.0 (3.6.1)
│   │   └── certifi (2024.7.4)
│   └── shapely>=1.8.0 (2.0.5)
│       └── numpy<3,>=1.14 (1.26.4)
├── h3>=4.0.0b1 (4.0.0b5)
├── osmnx>=1.3.0 (1.9.4)
│   ├── geopandas<0.15,>=0.12 (0.14.4)
│   │   ├── fiona>=1.8.21 (1.9.6)
│   │   │   ├── attrs>=19.2.0 (23.2.0)
│   │   │   ├── certifi (2024.7.4)
│   │   │   ├── click-plugins>=1.0 (1.1.1)
│   │   │   │   └── click>=4.0 (8.1.7)
│   │   │   ├── click~=8.0 (8.1.7)
│   │   │   ├── cligj>=0.5 (0.7.2)
│   │   │   │   └── click>=4.0 (8.1.7)
│   │   │   └── six (1.16.0)
│   │   ├── numpy>=1.22 (1.26.4)
│   │   ├── packaging (24.1)
│   │   ├── pandas>=1.4.0 (2.2.2)
│   │   │   ├── numpy>=1.22.4 (1.26.4)
│   │   │   ├── python-dateutil>=2.8.2 (2.9.0.post0)
│   │   │   │   └── six>=1.5 (1.16.0)
│   │   │   ├── pytz>=2020.1 (2024.1)
│   │   │   └── tzdata>=2022.7 (2024.1)
│   │   ├── pyproj>=3.3.0 (3.6.1)
│   │   │   └── certifi (2024.7.4)
│   │   └── shapely>=1.8.0 (2.0.5)
│   │       └── numpy<3,>=1.14 (1.26.4)
│   ├── networkx<3.4,>=2.5 (3.3)
│   ├── numpy<1.27,>=1.20 (1.26.4)
│   ├── pandas<2.3,>=1.1 (2.2.2)
│   │   ├── numpy>=1.22.4 (1.26.4)
│   │   ├── python-dateutil>=2.8.2 (2.9.0.post0)
│   │   │   └── six>=1.5 (1.16.0)
│   │   ├── pytz>=2020.1 (2024.1)
│   │   └── tzdata>=2022.7 (2024.1)
│   ├── requests<2.33,>=2.27 (2.32.3)
│   │   ├── certifi>=2017.4.17 (2024.7.4)
│   │   ├── charset-normalizer<4,>=2 (3.3.2)
│   │   ├── idna<4,>=2.5 (3.7)
│   │   └── urllib3<3,>=1.21.1 (2.2.2)
│   └── shapely<2.1,>=2.0 (2.0.5)
│       └── numpy<3,>=1.14 (1.26.4)
├── polars>=0.19.4 (1.2.1)
├── pooch>=1.6.0 (1.8.2)
│   ├── packaging>=20.0 (24.1)
│   ├── platformdirs>=2.5.0 (4.2.2)
│   └── requests>=2.19.0 (2.32.3)
│       ├── certifi>=2017.4.17 (2024.7.4)
│       ├── charset-normalizer<4,>=2 (3.3.2)
│       ├── idna<4,>=2.5 (3.7)
│       └── urllib3<3,>=1.21.1 (2.2.2)
├── psutil>=5.6.2 (6.0.0)
├── pyarrow-ops (0.0.8)
│   ├── numpy>=1.19.2 (1.26.4)
│   └── pyarrow>=3.0 (17.0.0)
│       └── numpy>=1.16.6 (1.26.4)
├── pyarrow>=16.0.0 (17.0.0)
│   └── numpy>=1.16.6 (1.26.4)
├── python-geohash>=0.8 (0.8.5)
├── requests (2.32.3)
│   ├── certifi>=2017.4.17 (2024.7.4)
│   ├── charset-normalizer<4,>=2 (3.3.2)
│   ├── idna<4,>=2.5 (3.7)
│   └── urllib3<3,>=1.21.1 (2.2.2)
├── rich>=10.11.0 (13.7.1)
│   ├── markdown-it-py>=2.2.0 (3.0.0)
│   │   └── mdurl~=0.1 (0.1.2)
│   └── pygments<3.0.0,>=2.13.0 (2.18.0)
├── s2>=0.1.9 (0.1.9)
│   ├── s2sphere (0.2.5)
│   │   └── future>=0.15 (1.0.0)
│   └── shapely (2.0.5)
│       └── numpy<3,>=1.14 (1.26.4)
├── shapely>=2 (2.0.5)
│   └── numpy<3,>=1.14 (1.26.4)
├── tqdm>=4.42.0 (4.66.4)
├── typeguard>=3.0.0 (4.3.0)
│   └── typing-extensions>=4.10.0 (4.12.2)
└── typer[all]>=0.9.0 (0.12.3)
    ├── click>=8.0.0 (8.1.7)
    ├── rich>=10.11.0 (13.7.1)
    │   ├── markdown-it-py>=2.2.0 (3.0.0)
    │   │   └── mdurl~=0.1 (0.1.2)
    │   └── pygments<3.0.0,>=2.13.0 (2.18.0)
    ├── shellingham>=1.3.0 (1.5.4)
    └── typing-extensions>=3.7.4.3 (4.12.2)
```

I'm currently working on eliminating the `OSMnx` dependency to replace it with `GeoPy`. It should be hopefully done in the near future.

Sigmanificient commented:
@RaczeQ Thanks for this clarification
I must admit that the dependency tree is quite huge, and i'm getting a bit lost :sweat_smile:

Sigmanificient commented:
Looks like they finally made the nixfmt formmatting mandatory

Sigmanificient commented:
Only missing dep for CLI is `s2`

Sigmanificient committed:
python313Packages.geoarrow-{c,pandas,pyarrow}: add myself in maintainers

Theses packages were orginally defined in this pr, so it makes sense for
me to also maintain them
